### ISSUE TITLE
Inconsistent image display with WandB frontend 

### ISSUE THREAD

**Issue by jramapuram** [wandb/client#1200]
wandb, version 0.9.4
Python 3.7.8
Linux

### Description

When uploading media from many different runs with different iterations / sec images are inconsistent.
Images are only shown with a browser refresh.

### What I Did

  - Have many runs going simultaneously with different iters/run.
  - Goto project page.
  - Images don't show up.
  - Set slider to first or second step --> no images shown. 
  - Images **only** shown correctly after refreshing.

See gif below:
![Screen Recording 2020-08-16 at 6 56 05 PM 2020-08-16 19_02_35](https://user-images.githubusercontent.com/8204807/90339796-88990c80-dff3-11ea-80c2-5ec28f0a5411.gif)

### Frontend Logs
```
:8080/files/jramapuram/kanerva_plus_plus_no_z/i8v7pmbb/media/images/test_reconstruction_12_22cfc909.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/adx6bkp2/media/images/test_reconstruction_22_ed0caa04.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/3f0jxqpr/media/images/test_reconstruction_16_9ba20931.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2gt0k2mn/media/images/test_reconstruction_7_6a8fd02c.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/8so5vq37/media/images/test_reconstruction_11_b0c9d326.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/pu4lm7md/media/images/test_reconstruction_8_56ed05b8.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/3gtcj6pq/media/images/test_reconstruction_11_67029558.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/1vq5lemw/media/images/test_reconstruction_6_2b672c82.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/1dtd39h1/media/images/test_reconstruction_11_aad9eb4b.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/ixzyks9z/media/images/test_reconstruction_19_d8753744.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2r6lbe50/media/images/test_reconstruction_16_523d8554.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/tug1bjn4/media/images/test_reconstruction_17_c2fd30c4.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2io9tyz1/media/images/test_reconstruction_13_acdfab98.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/6uv68jm8/media/images/test_reconstruction_17_fa191ad4.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2e1wu6r2/media/images/test_reconstruction_19_32a10501.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2gt0k2mn/media/images/train_mean-perturbed-generated_11_9b0fdd8f.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/pu4lm7md/media/images/train_mean-perturbed-generated_11_59b417e0.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/1vq5lemw/media/images/train_mean-perturbed-generated_10_1a183788.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/i8v7pmbb/media/images/train_mean-perturbed-generated_14_c8dc4542.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/8so5vq37/media/images/train_mean-perturbed-generated_14_9d7ce29e.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/3gtcj6pq/media/images/train_mean-perturbed-generated_14_e59280c7.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/i8v7pmbb/media/images/train_mean-perturbed-generated_17_d8155bb8.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/8so5vq37/media/images/train_mean-perturbed-generated_17_98045e09.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/3gtcj6pq/media/images/train_mean-perturbed-generated_16_67859d0f.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/1dtd39h1/media/images/train_mean-perturbed-generated_17_ed90538c.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2io9tyz1/media/images/train_mean-perturbed-generated_17_67cdd75e.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/adx6bkp2/media/images/train_mean-perturbed-generated_28_219c8268.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/ixzyks9z/media/images/train_mean-perturbed-generated_28_24b4127a.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/tug1bjn4/media/images/train_mean-perturbed-generated_19_13e41f42.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/2r6lbe50/media/images/train_mean-perturbed-generated_21_b948db24.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/adx6bkp2/media/images/train_mean-perturbed-generated_34_b956c09d.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kanerva_plus_plus_no_z/ixzyks9z/media/images/train_mean-perturbed-generated_29_bc94c486.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
DevTools failed to load SourceMap: Could not parse content for http://neuralnetworkart.com:8080/raven.min.js.map: Unexpected token < in JSON at position 0
```

**Comment by jramapuram** [comment]
To add on to this, from a different run it looks like the files are present on the server:
![image](https://user-images.githubusercontent.com/8204807/90394990-f8f76a80-e093-11ea-9240-adbfbd2230eb.png) 
but not all are present in the web artifacts UI:
![image](https://user-images.githubusercontent.com/8204807/90395042-13314880-e094-11ea-9a90-717ae15f7729.png)

There seems to be a disparity between resources present on disk and the frontend.

**Comment by vanpelt** [comment]
@jramapuram there may have been a persistence issue with our queueing logic.  For production deploys, we configure an external bucket and a persistent queue implementation.  In the trial version of wandb/local we run Redis to queue file events and if the instance gets shutdown before all files are processed they will not be present in the UI.  Let's confirm you're actually missing file records.  From within the container you can run:

```
sudo mysql -D wandb_local
SELECT name FROM files WHERE run_name = "3a38emrr";
```

Then we can compare that list with the files you have on disk.

**Comment by jramapuram** [comment]
Hi @vanpelt ,

So looks like the files did finally appear, but took quite a while to properly propagate:

```
+--------------------------------------------------------------+
| name                                                         |
+--------------------------------------------------------------+
| code/main.py                                                 |
| config.yaml                                                  |
| diff.patch                                                   |
| media/graph/graph_0_summary_77cb566f.graph.json              |
| media/images/test_input_10_62f959c1.png                      |
| media/images/test_input_11_62f959c1.png                      |
| media/images/test_input_1_62f959c1.png                       |
| media/images/test_input_2_62f959c1.png                       |
| media/images/test_input_3_62f959c1.png                       |
| media/images/test_input_4_62f959c1.png                       |
| media/images/test_input_5_62f959c1.png                       |
| media/images/test_input_6_62f959c1.png                       |
| media/images/test_input_7_62f959c1.png                       |
| media/images/test_input_8_62f959c1.png                       |
| media/images/test_input_9_62f959c1.png                       |
| media/images/test_mean-perturbed-generated_10_a091f148.png   |
| media/images/test_mean-perturbed-generated_11_4f2900c9.png   |
| media/images/test_mean-perturbed-generated_1_1910e144.png    |
| media/images/test_mean-perturbed-generated_2_4ce919f5.png    |
| media/images/test_mean-perturbed-generated_3_11ed31fc.png    |
| media/images/test_mean-perturbed-generated_4_b516adea.png    |
| media/images/test_mean-perturbed-generated_5_9532d1ce.png    |
| media/images/test_mean-perturbed-generated_6_ad080a13.png    |
| media/images/test_mean-perturbed-generated_7_5354a9d0.png    |
| media/images/test_mean-perturbed-generated_8_635a631a.png    |
| media/images/test_mean-perturbed-generated_9_320f06ba.png    |
| media/images/test_memory_10_a26b2dda.png                     |
| media/images/test_memory_11_8797d008.png                     |
| media/images/test_memory_1_4fbc07e0.png                      |
| media/images/test_memory_2_fa8b6ddb.png                      |
| media/images/test_memory_3_47665165.png                      |
| media/images/test_memory_4_8e17d543.png                      |
| media/images/test_memory_5_9db4e290.png                      |
| media/images/test_memory_6_13b0040f.png                      |
| media/images/test_memory_7_418c0bc7.png                      |
| media/images/test_memory_8_838624ff.png                      |
| media/images/test_memory_9_cea43253.png                      |
| media/images/test_mosiac-generated_10_a78ecc08.png           |
| media/images/test_mosiac-generated_11_b7817536.png           |
| media/images/test_mosiac-generated_1_1910e144.png            |
| media/images/test_mosiac-generated_2_3ea21f39.png            |
| media/images/test_mosiac-generated_3_80aed177.png            |
| media/images/test_mosiac-generated_4_15848765.png            |
| media/images/test_mosiac-generated_5_552d10fc.png            |
| media/images/test_mosiac-generated_6_987ccc19.png            |
| media/images/test_mosiac-generated_7_463e5004.png            |
| media/images/test_mosiac-generated_8_04c4c691.png            |
| media/images/test_mosiac-generated_9_16783e15.png            |
| media/images/test_read-traces-minibatch0_10_995fec55.png     |
| media/images/test_read-traces-minibatch0_11_d21bc3a3.png     |
| media/images/test_read-traces-minibatch0_1_971c8d94.png      |
| media/images/test_read-traces-minibatch0_2_642292da.png      |
| media/images/test_read-traces-minibatch0_3_d9c71b45.png      |
| media/images/test_read-traces-minibatch0_4_cc66a5b5.png      |
| media/images/test_read-traces-minibatch0_5_e64aebc0.png      |
| media/images/test_read-traces-minibatch0_6_273cb434.png      |
| media/images/test_read-traces-minibatch0_7_c11f3bf0.png      |
| media/images/test_read-traces-minibatch0_8_fc858dd1.png      |
| media/images/test_read-traces-minibatch0_9_631b287f.png      |
| media/images/test_reconstruction_10_baadce9f.png             |
| media/images/test_reconstruction_11_1f9d9273.png             |
| media/images/test_reconstruction_1_d2c571aa.png              |
| media/images/test_reconstruction_2_b162f785.png              |
| media/images/test_reconstruction_3_d932e33c.png              |
| media/images/test_reconstruction_4_2235eb05.png              |
| media/images/test_reconstruction_5_af99612c.png              |
| media/images/test_reconstruction_6_deb2c929.png              |
| media/images/test_reconstruction_7_09704b41.png              |
| media/images/test_reconstruction_8_33edfd7b.png              |
| media/images/test_reconstruction_9_63f1ef8a.png              |
| media/images/train_input_10_0fa343c8.png                     |
| media/images/train_input_11_95f51c82.png                     |
| media/images/train_input_1_34a1d5d4.png                      |
| media/images/train_input_2_d046162f.png                      |
| media/images/train_input_3_7d86d1b4.png                      |
| media/images/train_input_4_ca5131c8.png                      |
| media/images/train_input_5_8dd3d680.png                      |
| media/images/train_input_6_9317c0a1.png                      |
| media/images/train_input_7_8ce9cf95.png                      |
| media/images/train_input_8_d8007113.png                      |
| media/images/train_input_9_6463d8e2.png                      |
| media/images/train_mean-perturbed-generated_10_b54b85b4.png  |
| media/images/train_mean-perturbed-generated_11_0fb961b4.png  |
| media/images/train_mean-perturbed-generated_1_475a50ea.png   |
| media/images/train_mean-perturbed-generated_2_e1b62153.png   |
| media/images/train_mean-perturbed-generated_3_3f28ed32.png   |
| media/images/train_mean-perturbed-generated_4_b39799d4.png   |
| media/images/train_mean-perturbed-generated_5_fa1efcba.png   |
| media/images/train_mean-perturbed-generated_6_23fdb778.png   |
| media/images/train_mean-perturbed-generated_7_1c1416ee.png   |
| media/images/train_mean-perturbed-generated_8_e5782731.png   |
| media/images/train_mean-perturbed-generated_9_2fa52b4e.png   |
| media/images/train_memory_10_33ef80c0.png                    |
| media/images/train_memory_11_2db84136.png                    |
| media/images/train_memory_1_661dbcf0.png                     |
| media/images/train_memory_2_6a6c5af3.png                     |
| media/images/train_memory_3_cbb0a6e6.png                     |
| media/images/train_memory_4_225be40b.png                     |
| media/images/train_memory_5_5ca3120c.png                     |
| media/images/train_memory_6_be473a2f.png                     |
| media/images/train_memory_7_871cc013.png                     |
| media/images/train_memory_8_d6c58600.png                     |
| media/images/train_memory_9_ffab958f.png                     |
| media/images/train_mosiac-generated_10_ca68604c.png          |
| media/images/train_mosiac-generated_11_6411da20.png          |
| media/images/train_mosiac-generated_1_475a50ea.png           |
| media/images/train_mosiac-generated_2_0818030b.png           |
| media/images/train_mosiac-generated_3_b5a23425.png           |
| media/images/train_mosiac-generated_4_a1c61d0c.png           |
| media/images/train_mosiac-generated_5_e0b4b3f8.png           |
| media/images/train_mosiac-generated_6_3fb18cdf.png           |
| media/images/train_mosiac-generated_7_bb473dc7.png           |
| media/images/train_mosiac-generated_8_5a899f5a.png           |
| media/images/train_mosiac-generated_9_b1605565.png           |
| media/images/train_read-traces-minibatch0_10_d3b4dc0e.png    |
| media/images/train_read-traces-minibatch0_11_a816032d.png    |
| media/images/train_read-traces-minibatch0_1_486c7452.png     |
| media/images/train_read-traces-minibatch0_2_2707d407.png     |
| media/images/train_read-traces-minibatch0_3_60daf4ad.png     |
| media/images/train_read-traces-minibatch0_4_f995b459.png     |
| media/images/train_read-traces-minibatch0_5_2c3b1d39.png     |
| media/images/train_read-traces-minibatch0_6_f7044779.png     |
| media/images/train_read-traces-minibatch0_7_7fd2cb43.png     |
| media/images/train_read-traces-minibatch0_8_3305d885.png     |
| media/images/train_read-traces-minibatch0_9_519f09c0.png     |
| media/images/train_reconstruction_10_1af3ed44.png            |
| media/images/train_reconstruction_11_c44ccb40.png            |
| media/images/train_reconstruction_1_fc2a7fa1.png             |
| media/images/train_reconstruction_2_4e92529f.png             |
| media/images/train_reconstruction_3_d291b25e.png             |
| media/images/train_reconstruction_4_e10eeb54.png             |
| media/images/train_reconstruction_5_f3eb84dc.png             |
| media/images/train_reconstruction_6_f3f80736.png             |
| media/images/train_reconstruction_7_d8d734bf.png             |
| media/images/train_reconstruction_8_e95ba55b.png             |
| media/images/train_reconstruction_9_fa6d8d42.png             |
| output.log                                                   |
| requirements.txt                                             |
| upstream_diff_3d9e8cf427f8b8d1868fa378e64a5ca4c8afdfe9.patch |
| wandb-metadata.json                                          |
| wandb-summary.json                                           |
+--------------------------------------------------------------+
141 rows in set (0.03 sec)
```

![image](https://user-images.githubusercontent.com/8204807/90420260-01619c80-e0b8-11ea-8d56-635ddb835e9f.png)

Is this to be expected? With Visdom I see results instantly propagated.

**Comment by vanpelt** [comment]
They should be showing up within seconds unless the instance is overloaded or under provisioned.  Can you check the system load inside the docker container next time you're running runs?  How much RAM / CPU does the instance have?  The suggested minimum is 4 CPU's and 8GB of RAM.

**Comment by jramapuram** [comment]
I'm currently running it on a 12core system with 80Gb of RAM and the file-system is backed with a Raid-0 SSD. `htop` reveals no major CPU / RAM usage issues.
![image](https://user-images.githubusercontent.com/8204807/90420991-08d57580-e0b9-11ea-8528-afa14b47147e.png)

**Comment by vanpelt** [comment]
How many images are you logging per second in the run?  Within the instance we can see them getting processed in the log: `/var/log/gorilla-filemeta.log`.  The python process asks our api for an upload url, then uploads the file to the Minio instance running in the VM.  Minio then sends an event to a redis queue and the gorilla-filemeta process consumes events from this queue and writes them to the DB.  It sounds like this queue is getting backed up for some reason, hopefully the log get's us more insight.

**Comment by jramapuram** [comment]
I tried running around 8 experiments simultaneously with each logging around 12 images per 10-20s. I wouldn't expect that to saturate the network or disk IO.

The logs you requested ( I didn't see anything obviously wrong there; I didn't a log-rotate version of these logs either ) :
https://gist.githubusercontent.com/jramapuram/fb081343b32ef993e0074af619098fcd/raw/775847ff969ecee49f602d84c414b99dac900dae/gorilla-filemeta.log

I think your original speculation might be right with the desync'd ordering.

**Comment by vanpelt** [comment]
Strange.  If you can send a simple script the reproduces the issue we can debug on our end and see what's going on.

**Comment by jramapuram** [comment]
I'll try to setup a minimal reproducible sample when I get some time and post back here.

**Comment by jramapuram** [comment]
I just had this happen with just 1 running experiment in local mode.
![image](https://user-images.githubusercontent.com/8204807/92536617-6c773c80-f23a-11ea-8148-bd8bdf5ddb24.png)

```
:8080/files/jramapuram/kpp3d_single_mem/12qla2f4/media/images/test_mosiac-generated0_343_5c4d83b1.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kpp3d_single_mem/12qla2f4/media/images/test_mean-perturbed-generated5_352_0f25651a.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kpp3d_single_mem/12qla2f4/media/images/test_mean-perturbed-generated7_356_5bd8859a.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
:8080/files/jramapuram/kpp3d_single_mem/12qla2f4/media/images/test_input_341_9909e173.png:1 Failed to load resource: the server responded with a status of 404 (Not Found)
```

**Comment by cvphelps** [comment]
Hi there,I believe this is fixed in our latest release `0.10.x`. Could you please kindly try the update and reopen this ticket if it's still a problem.
